### PR TITLE
Adding Great Expectations base operator

### PR DESCRIPTION
**Pull request by spbail**
Making a PR against our fork so we can have an internal review first.

How to test this: 

- Checkout my branch sam/airflow_demo on https://github.com/superconductive/ge_demo/pull/32/files 
- Adjust the paths in `ge_dag.py` accordingly
- Clone this airflow fork and install with `pip install -e .`
- You'll probably have to nuke your airflow.db (most likely in `~/airflow/airflow.db` and rerun `airflow db init` to migrate to airflow 2.0
- Then in the ge_demo repo run `airflow tasks test ge_dag ge_checkpoint_pass_wrong_root_dir 2020-01-01` (or any of the other task names in ge_dag.py), they'll either pass or fail (the name will indicate)
- Feel free to add your own tasks to ge_dag.py to experiment with different parameters

### CODE REVIEW

File: airflow/providers/greatexpectations/operators/greatexpectations.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import great_expectations as ge
+
+log = logging.getLogger(__name__)
+
+
+class GreatExpectationsOperator(BaseOperator):
+    @apply_defaults
+    def __init__(self,
+                 *,
+                 run_id=None,
```
Comment (jcampbell) [inline airflow/providers/greatexpectations/operators/greatexpectations.py:34]: Update to the new run_name; that will then be paired with the run time automatically.

```suggestion
                 run_name=None,
```

File: airflow/providers/greatexpectations/operators/greatexpectations.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+                 checkpoint_name=None,
+                 fail_task_on_validation_failure=True,
+                 validation_operator_name="action_list_operator",
+                 **kwargs
+                 ):
+        """
+        Args:
+            run_id: Optional run_id to identify the validation run (defaults to timestamp if not specified)
+            data_context_root_dir: Path of the great_expectations directory data_context: A great_expectations
+            DataContext object expectation_suite_name: The name of the Expectation Suite to use for validation
```
Comment (jcampbell) [inline airflow/providers/greatexpectations/operators/greatexpectations.py:49]: I think formatting got kerfuffled

File: airflow/providers/greatexpectations/operators/greatexpectations.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            batch_kwargs: The batch_kwargs to use for validation assets_to_validate: A list of dictionaries of
+            batch_kwargs + expectation suites to use for validation checkpoint_name: A Checkpoint name to use for
+            validation fail_task_on_validation_failure: Fail the Airflow task if the Great Expectation validation fails
+            validation_operator_name: Optional name of a Great Expectations validation operator, defaults to
+            action_list_operator
+            **kwargs: Optional kwargs
+        """
+        super().__init__(**kwargs)
+
+        self.run_id = run_id
```
Comment (jcampbell) [inline airflow/providers/greatexpectations/operators/greatexpectations.py:59]:
```suggestion
        self.run_name = run_name
```

File: airflow/providers/greatexpectations/operators/greatexpectations.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+                batch = self.data_context.get_batch(
+                    asset["batch_kwargs"],
+                    asset["expectation_suite_name"]
+                )
+                batches_to_validate.append(batch)
+
+        results = self.data_context.run_validation_operator(
+            validation_operator_name,
+            assets_to_validate=batches_to_validate,
+            run_id=self.run_id
```
Comment (jcampbell) [inline airflow/providers/greatexpectations/operators/greatexpectations.py:120]:
```suggestion
            run_name=self.run_name
```

**Review by jcampbell — CHANGES_REQUESTED** [review]
<3!

**Review by eugmandel — COMMENTED** [review]
Looks good to me. Agree with @jcampbell 's run_name comment. I went over the code, but I cannot try it yet, because pip-installing the Airflow fork hangs. I'll try resolving it.

File: airflow/providers/greatexpectations/operators/greatexpectations.py (hunk 2 above)
Comment (spbail) [reply airflow/providers/greatexpectations/operators/greatexpectations.py:49]: Ah yeah, fixed it.

**Review by spbail — COMMENTED** [review]
Changed the run_id to run_name and fixed the docstrings

**Pull request merged**